pinot-minion/src/main/java/org/apache/pinot/minion/BaseMinionStarter.java

```diff
@@ -281,6 +281,7 @@ public void start()
     minionContext.setHelixManager(_helixManager);
     LOGGER.info("Starting minion admin application on: {}", ListenerConfigUtil.toString(_listenerConfigs));
     _minionAdminApplication = new MinionAdminApiApplication(_instanceId, _config);
+    registerExtraComponents(_minionAdminApplication);
```

Comment:
Suggest modeling it as `createMinionAdminApp()` which returns `MinionAdminApiApplication` and can be overridden for flexibility. This way we can also support extending `MinionAdminApiApplication` if necessary. Same for other components.

Reply:
For broker, we can consider adding this new method as well, and mark `registerExtraComponents()` deprecated

Reply:
Added in the latest commit.